src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -509,6 +507,12 @@ calculate_norm_matrix(TargetT &normp,
       const int min_y = emission[z].get_min_index();
       const int max_y = emission[z].get_max_index();
       this->dimy=max_y-min_y+1;
+
+#ifdef STIR_OPENMP
+#pragma omp parallel
+    {
+      #pragma omp  for schedule(dynamic) private(l,m)//schedule(static,1)
```

Comment:
syntax ok I believe, but shorter to just say
```c++
#pragma omp parallel for ....
```

Reply:
Also, as you declare `l,m` private, it means that their values outside of the loop will be unpredictable. That makes me wonder if they shouldn't be declared locally below, i.e.
```c++
const int l = (z-min_z)*(max_x-min_x +1)*(max_y-min_y +1)
                + (y-min_y)*(max_x-min_x +1) + (x-min_x);
```
far cleaner...

Reply:
Finally, you can parallelise nested for loops as in https://stackoverflow.com/questions/13357065/how-does-openmp-handle-nested-loops.
that might help with speeding up if you have a lot of cores.

However, I think this is only since openmp 3.1 (do check). Visual Studio is still on OpenMP 2.0... Interestingly, OpenMP versions are encoded via date, see https://stackoverflow.com/questions/1304363/how-to-check-the-version-of-openmp-on-linux/13612520
So, I'm afraid you'll have to do something like
```c++
#ifdef STIR_OPENMP
#  if _OPENMP <201107
#    pragma omp  for schedule(dynamic)
#  else
#    pragma omp  for schedule(dynamic) collapse(2)
#  endif
#endif
```

Reply:
I had tried to use nested but I get errors! It seems that only works if there is nothing between the consecutive loops, ex:
For(){
   For(){
     For(){}}}

Reply:
ah well... an alternative reason might be that in the code min_x depends on y. that probably makes it too hard for the compiler anyway.
One solution would be to require "rectangular" images, and compute min_z/y/z etc upfront (e.g. with get_regular_range). I suspect that this is required for your code anyway.
you probably would still have to move min_dy etc inside but as that's a near trivial calculation, that's not going to do any harm.

Reply:
> you probably would still have to move min_dy etc inside but as that's a near trivial calculation, that's not going to do any harm.

Shall I then move min_dz inside so that I can do collapse(3)?